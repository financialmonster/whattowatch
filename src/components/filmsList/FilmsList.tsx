import React, { FC, memo } from 'react';
import { List, Map } from 'immutable';

import { Film } from 'components/film/Film';

type TFilmsListProps = {
    films: List<Map<string, any>> | undefined;
}

export const FilmsList: FC<TFilmsListProps> = memo(({ films }) => {
    const filmsJSX: List<JSX.Element> =
        (films as List<Map<string, any>>).map((film) => <Film key={film.get(`id`)} film={film} />);

    return (
        <div className="catalog__movies-list">
            {filmsJSX}
        </div>
    );
});
