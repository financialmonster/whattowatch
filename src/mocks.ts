import { Map, fromJS, List } from 'immutable';

export const createStoreMock = (mode: string = `common`) => {
    return {
        promo: Map({
            promo: Map(),
            isPromoFetching: true,
            promoError: null
        }),
        films: Map({
            films: fromJS([{genre: `Thriller`, id: 1}, {genre: `Comedy`, id: 2}]),
            isFilmsFetching: mode === `detailed`,
            filmsError: null,
            filter: `All genres`
        }),
        auth: Map({
            user: null,
            authError: null,
            isAuthFetching: false,
            authStatusError: null,
            isAuthStatusFetching: false
        }),
        reviews: Map({
            reviews: List(),
            isReviewsFetching: true,
            reviewsError: null
        })
    }
}
