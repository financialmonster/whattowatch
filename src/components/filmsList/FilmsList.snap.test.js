import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import { FilmsList } from './FilmsList';
import { createStoreMock } from 'mocks';

it(`FilmsList: correctly rendered`, () => {
    const mockStore = configureStore();
    const store = mockStore(createStoreMock());

    const tree = renderer.create(
        <Provider store={store}>
            <FilmsList films={store.getState().films.get('films')} />
        </Provider>   
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
