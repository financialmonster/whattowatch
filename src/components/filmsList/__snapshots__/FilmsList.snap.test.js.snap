// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmsList: correctly rendered 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
    >
      <img
        height="175"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
      />
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
    >
      <img
        height="175"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
      />
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
    >
      <img
        height="175"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
      />
    </h3>
  </article>
</div>
`;
